GitHubContentUtility/Operations/BlobContentWriter.cs

```diff
@@ -0,0 +1,163 @@
+﻿// ------------------------------------------------------------------------------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------------------------------------------------------------------------------
+
+using Octokit;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using GitHubContentUtility.Common;
+using GitHubContentUtility.Services;
+
+namespace GitHubContentUtility.Operations
+{
+    /// <summary>
+    /// Provides methods for writing contents to a specific blob in a GitHub repository.
+    /// </summary>
+    public class BlobContentWriter
+    {
+        /// <summary>
+        /// Writes contents to a specified blob in the specified GitHub repository.
+        /// </summary>
+        /// <param name="appConfig">The application configuration object which contains values
+        /// for connecting to the specified GitHub repository.</param>
+        /// <param name="privateKey"> The RSA private key of a registered GitHub app installed in the specified repository.</param>
+        /// <returns>A task.</returns>
+        public async Task WriteToRepositoryAsync(ApplicationConfig appConfig, string privateKey)
+        {
+            if (appConfig == null)
+            {
+                throw new ArgumentNullException(nameof(appConfig), "Parameter cannot be null");
+            }
+
+            try
+            {
+                string token = await GitHubAuthService.GetGithubAppTokenAsync(appConfig, privateKey);
+
+                // Pass the JWT as a bearer token to Octokit.net
+                var finalClient = new GitHubClient(new ProductHeaderValue(appConfig.GitHubAppName))
+                {
+                    Credentials = new Credentials(token, AuthenticationType.Bearer)
+                };
+
+                // Get repo references
+                var references = await finalClient.Git.Reference.GetAll(appConfig.GitHubOrganization, appConfig.GitHubRepoName);
+
+                // Check if the working branch is in the refs
+                var workingBranch = references.Where(reference => reference.Ref == $"refs/heads/{appConfig.WorkingBranch}").FirstOrDefault();
+
+                // Check if branch already exists
+                if (workingBranch == null)
+                {
+                    // Working branch does not exist, so branch off from the reference branch
+                    var refBranch = references.Where(reference => reference.Ref == $"refs/heads/{appConfig.ReferenceBranch}").FirstOrDefault();
```

Comment:
probably need to null check this since the value is passed in

Reply:
Since there are many `appConfig` properties that would need null-checking, I saw it best to allow an exception to be caught and thrown in scenarios of nullability.